### PR TITLE
JS implementation for SendValue command

### PR DESCRIPTION
**Pull request by colombod**
Value sharing is now handled on the js kernel
image

the kernel with `#!share` is not listed as a completion for the `--from` option
image

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive/Commands/SendValue.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class SendValue : KernelCommand
         FormattedValue = formattedValue;
     }
 
-    public FormattedValue FormattedValue { get; }
+    public FormattedValue FormattedValue { get; internal set; }
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Commands/SendValue.cs:27]: Why is this needed?

File: src/Microsoft.DotNet.Interactive/ValueSharing/JavaScriptValueDeclarer.cs [hunk 2]
```diff
@@ -34,21 +33,18 @@ static JavaScriptValueDeclarer()
...
-        out SubmitCode command)
+    public static bool TryGetValueDeclaration(object referenceValue, string declareAsName, out string code)
     {
-        if (valueProduced.Value is { } value)
+        if (referenceValue is { } value)
         {
-            var code = $"{valueProduced.Name} = {JsonSerializer.Serialize(value, _serializerOptions)};";
-            command = new SubmitCode(code);
+            code = $"{declareAsName} = {JsonSerializer.Serialize(value, _serializerOptions)};";
+            new SubmitCode(code);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/ValueSharing/JavaScriptValueDeclarer.cs:41]: This line can go.

File: src/microsoft-dotnet-interactive/src/javascriptKernel.ts [hunk 3]
```diff
@@ -11,15 +11,34 @@ export class JavascriptKernel extends Kernel {
...
     }
 
+    private handleSendValue(invocation: IKernelCommandInvocation): Promise<void> {
+        const sendValue = <contracts.SendValue>invocation.commandEnvelope.command;
+        if (sendValue.formattedValue) {
+            switch (sendValue.formattedValue.mimeType) {
+                case 'application/json':
+                    (<any>globalThis)[sendValue.name] = JSON.parse(sendValue.formattedValue.value);
+                    break;
+                case 'text/plain':
```
Comment (jonsequitur) [inline src/microsoft-dotnet-interactive/src/javascriptKernel.ts:31]: This MIME type feels wrong to me. Maybe `text/javascript`?

File: src/Microsoft.DotNet.Interactive/Commands/SendValue.cs (hunk 1 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive/Commands/SendValue.cs:27]: No reason

File: src/Microsoft.DotNet.Interactive/ValueSharing/JavaScriptValueDeclarer.cs (hunk 2 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive/ValueSharing/JavaScriptValueDeclarer.cs]: done

File: src/Microsoft.DotNet.Interactive/Commands/SendValue.cs (hunk 1 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive/Commands/SendValue.cs]: 🤨

File: src/microsoft-dotnet-interactive/src/javascriptKernel.ts (hunk 3 above)
Comment (colombod) [reply src/microsoft-dotnet-interactive/src/javascriptKernel.ts:31]: if the variable is shared as plain/text we trat that as string value. what should be the behaviour ?

Comment (jonsequitur) [reply src/microsoft-dotnet-interactive/src/javascriptKernel.ts:31]: I'm not sure this MIME type should be valid, or else we should just declare a string.

Comment (colombod) [reply src/microsoft-dotnet-interactive/src/javascriptKernel.ts:31]: right now if is text/plain it declares a string.

Comment (colombod) [reply src/microsoft-dotnet-interactive/src/javascriptKernel.ts]: Deleted that case

File: src/Microsoft.DotNet.Interactive.Tests/VariableSharingTests.cs [hunk 4]
```diff
@@ -542,7 +543,7 @@ await localCompositeKernel
                           remoteKernelUri);
 
             javascriptKernel.UseValueSharing();
-
+            javascriptKernel.RegisterCommandType<SendValue>();
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/VariableSharingTests.cs:546]: It shouldn't be necessary to register this on both the remote and the proxy. Is there a bug here?

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Tests/VariableSharingTests.cs:546]: good point, not needed

File: src/Microsoft.DotNet.Interactive.Browser.Tests/JavaScriptKernelTests.cs [hunk 5]
```diff
@@ -72,6 +73,64 @@ public async Task It_can_get_console_log_output()
                                   v.Value == "123");
     }
 
+    [FactSkipLinux]
```
[the remark this answers is not in the document]
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Browser.Tests/JavaScriptKernelTests.cs]: I have added the ability to give a reason why we skip.

**Review by brettfo — APPROVED** [review]
